projects/sunbird-video-player/src/lib/services/viewer.service.spec.ts

```diff
@@ -1,14 +1,78 @@
   }));
 
-  it('should be created', () => {
-    const service: ViewerService = TestBed.get(ViewerService);
+  it('should be created ViewerService', () => {
+    const service = TestBed.inject(ViewerService);
     expect(service).toBeTruthy();
   });
+  it('should call raiseExceptionLog', () => {
+    const service = TestBed.inject(ViewerService);
+    spyOn(service.playerEvent, 'emit').and.callThrough();
+    spyOn(service['videoPlayerService'], 'error').and.callFake(() => 'true');
+    const exceptionLogEvent = {
+      eid: 'ERROR',
+      edata: {
+        err: 'errorCode',
+        errtype: 'errorType',
+        requestid: 'traceId',
+        stacktrace: 'stacktrace',
+      }
+    };
+    service.raiseExceptionLog(exceptionLogEvent.edata.err,
+      exceptionLogEvent.edata.errtype, exceptionLogEvent.edata.stacktrace, exceptionLogEvent.edata.requestid);
+    expect(service.playerEvent.emit).toHaveBeenCalledWith(exceptionLogEvent);
+    expect(service['videoPlayerService']['error']).toHaveBeenCalledWith(exceptionLogEvent.edata.err,
+      exceptionLogEvent.edata.errtype, exceptionLogEvent.edata.stacktrace);
+  });
+  it('should call raiseHeartBeatEvent for REPLAY', () => {
+    const service = TestBed.inject(ViewerService);
+    spyOn(service.playerEvent, 'emit').and.callThrough();
+    spyOn(service['videoPlayerService'], 'heartBeat').and.callFake(() => 'true');
+    spyOn(service['videoPlayerService'], 'interact').and.callFake(() => 'true');
+    service.raiseHeartBeatEvent('REPLAY');
+    expect(service.showScore).toBeFalsy();
+    expect(service.scoreObtained).toEqual(0);
+    expect(service.interceptionResponses).toEqual({});
+  });
+  it('should call calculateScore', () => {
+    const service = TestBed.inject(ViewerService);
+    service.interceptionResponses = {};
+    service.calculateScore();
+    expect(service.scoreObtained).toEqual(0);
+  });
+  it('should call raiseHeartBeatEvent for REPLAY', () => {
+    const service = TestBed.inject(ViewerService);
+    spyOn(service.playerEvent, 'emit').and.callThrough();
+    spyOn(service['videoPlayerService'], 'start').and.callFake(() => 'true');
+    service.raiseStartEvent('');
+    expect(service.PlayerLoadStartedAt).toBeDefined();
+  });
+  it('should call preFetchContent', () => {
+    const service = TestBed.inject(ViewerService);
+    spyOn(service, 'getNextMarker').and.returnValue({identifier: '1234'});
+    spyOn(service, 'getQuestionSet').and.callFake(() => 'true');
+    service.preFetchContent();
+    expect(service.getQuestionSet).toHaveBeenCalledWith('1234');
+  });
+  it('should call raiseEndEvent', () => {
+    const service = TestBed.inject(ViewerService);
+    service.isEndEventRaised = true;
+    spyOn(service, 'calculateScore').and.callThrough();
+    spyOn(service['utilService'], 'getTimeSpentText').and.callFake(() => 'true');
+    service.raiseEndEvent();
+    expect(service['utilService'].getTimeSpentText).not.toHaveBeenCalledWith(10);
```

Comment:
Check this value **toHaveBeenCalledWith(10);**

Reply:
Now mocking **visitedLength**  value  and passing that value to getTimeSpentText(visitedLength)